client/pages/home.jsx

```diff
@@ -1,8 +1,206 @@
+    this.setState({
+      [event.target.name]: value
+    }, this.calculateNights);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    };
+    fetch('/api/trip', req)
+      .then(response => response.json())
+      .then(newTrip => {
+        const newTripEntriesArray = this.props.tripEntries.concat(newTrip);
+        this.props.homeState({ tripEntries: newTripEntriesArray });
+        event.target.reset();
+      })
+      .catch(err => console.error(err));
+
+    this.props.homeState({
+      modalView: false
+    });
+  }
+
+  handleClickOff(e) {
+    this.props.homeState({
+      modalView: false
+    });
+  }
+
+  stopPropagation(e) {
+    e.stopPropagation();
+  }
+
+  render() {
+    const today = new Date();
+    let month = today.getMonth() + 1;
+    let day = today.getDate() + 1;
+    const year = today.getFullYear();
+    if (month < 10) {
+      month = '0' + month.toString();
+    }
+    if (day < 10) {
+      day = '0' + day.toString();
+    }
+    const maxDate = `${year}-${month}-${day}`;
+
+    let returnMaxDate;
+    let disabled;
+    if (!this.state.departureDate) {
+      returnMaxDate = maxDate;
+      disabled = true;
+    } else {
+      returnMaxDate = this.state.departureDate;
+      disabled = false;
+    }
+
+    if (!this.props.view) {
+      return null;
+    } else {
+      return (
+        <div className="overlay" onClick={this.handleClickOff}>
+          <div className="trip-modal" onClick={this.stopPropagation}>
+            <h2>Add a trip</h2>
+            <form id="tripForm" onSubmit={this.handleSubmit}>
+              <label htmlFor="tripName">Name of trip</label>
+              <input type="text" id="tripName" name="tripName" onChange={this.handleChange} value={this.state.tripName} required></input>
+              <label htmlFor="tripDestination">Destination</label>
+              <input type="text" id="tripDestination" name="tripDestination" onChange={this.handleChange} value={this.state.tripDestination} required></input>
+              <label htmlFor="depatureDate">Depature Date</label>
+              <input type="date" id="departureDate" name="departureDate" min={maxDate} onChange={this.handleChange} value={this.state.departureDate} required></input>
+              <label htmlFor="returnDate">Return Date</label>
+              <input disabled={disabled} type="date" id="returnDate" name="returnDate" min={returnMaxDate} onChange={this.handleChange} value={this.state.returnDate} required></input>
+              <button className="rounded-lg tripsubmit" type="submit" value="Submit">Add this trip</button>
+            </form>
+          </div>
+        </div>
+      );
+    }
+  }
+}
+
+// function TripEntries(props) {
+//   return (
+//     <div className="container">
+//       <h1>Your Trips</h1>
+//     </div>
+//   );
+// }
```

Comment:
## 👮‍♀️

Reply:
Had to stop myself. Realized I was going into the next feature.